packages/ui/src/lib/modal/Modal.svelte

```diff
@@ -6,12 +6,14 @@ import { tabWithinParent } from '../utils/dialog-utils';
 const dispatch = createEventDispatcher();
 
 let modal: HTMLDivElement;
-export let name = 'drop-down-dialog';
-export let top: boolean = false;
-export let ignoreFocusOut: boolean = false;
-export let onclose: () => void = () => {
-  dispatch('close');
-};
+let {
+  name = 'drop-down-dialog',
+  top = false,
+  ignoreFocusOut = false,
+  onclose = (): void => {
+    dispatch('close');
+  },
+} = $props();
```

Comment:
issue: use a Props interface with the type of the objects

Reply:
Is this required ? Can get the added value

Reply:
Here yes, as typing is not explicit otherwise

Reply:
Isn't each deconstructed member typed here ?

Reply:
https://svelte.dev/docs/svelte/$props#Type-safety

Reply:
here, you're assigning default values, not adding the type